src/REveMu2eMainWindow.cc

```diff
@@ -252,84 +149,233 @@ void REveMu2eMainWindow::GeomDrawer(TGeoNode* node, REX::REveTrans& trans, REX::
-    for(auto& i: substring_tracker){
-      shift.at(0) = 0;
-      shift.at(1) = 0;
-      shift.at(2) = 0 ;
-      showNodesByName(node,i,kFALSE, 0, trans, trackerholder, maxlevel, level, false, false, shift, false, true, drawconfigf.getInt("TRKColor"));
-    }
-  }if(geomOpt.showST and !geomOpt.extracted){
-    static std::vector <std::string> substrings_stoppingtarget  {"TargetFoil"};
-    shift.at(0) = 0;
-    shift.at(1) = 0;
-    shift.at(2) = -1*STz/10 - 6.5; // temporary kludge
-    for(auto& i: substrings_stoppingtarget){
-      showNodesByName(node,i,kFALSE, 0, trans, beamlineholder, maxlevel, level, false, false, shift, false, true, drawconfigf.getInt("BLColor"));
+    // everything else needs to be shifted such that its relative to the tracker center at 0,0,0
+    if(geomOpt.showST and !geomOpt.extracted){
+      static std::vector <std::string> substrings_stoppingtarget  {"TargetFoil"};
+      shift.at(0) = x_st - x_trk;
+      shift.at(1) = y_st - y_trk;
+      shift.at(2) = z_st - z_trk;
+      for(auto& i: substrings_stoppingtarget){
+        showNodesByName(node,i,kFALSE, 0, trans, targetholder, maxlevel, level, false, false, shift, false, true, drawconfigf.getInt("BLColor"));
+      }
     }
-  }
-  if(geomOpt.showST and !geomOpt.extracted){
-    static std::vector <std::string> substrings_stoppingtarget  {"protonabs"};
-    shift.at(0) = 0;
-    shift.at(1) = 0;
-    shift.at(2) = -1*STz/10 + 32; // temporary kludge
-    for(auto& i: substrings_stoppingtarget){
-      showNodesByName(node,i,kFALSE, 0, trans, targetholder, maxlevel, level, false, false, shift, false, true, drawconfigf.getInt("BLColor"));
+    if(geomOpt.showCalo){
+        static std::vector <std::string> substrings_disk  {"caloDisk"};
+        for(auto& i: substrings_disk){
+          shift.at(0) = x_cal - x_trk;
+          shift.at(1) = y_cal - y_trk;
+          shift.at(2) = z_cal - z_trk;
+          showNodesByName(node,i,kFALSE, 0, trans, caloholder, maxlevel, level, true, false, shift, false, false, drawconfigf.getInt("CALColor") );
+        }
+      if(geomOpt.showCaloCrystals){
+        static std::vector <std::string> substrings_crystals  {"caloCrystal"};
+        for(auto& i: substrings_crystals){
+          shift.at(0) = x_cal - x_trk;
+          shift.at(1) = y_cal - y_trk;
+          shift.at(2) = z_cal - z_trk;
+          showNodesByName(node,i,kFALSE, 0, trans, crystalsholder, maxlevel, level, true, true, shift, false, false, drawconfigf.getInt("CALColor"));
+        }
+      }
     }
-  }
-
-  if(geomOpt.showSTM){
-    static std::vector <std::string> substrings_stm  {"stmDet1Can","stmDet1","stmDet2Can","stmDet2"};
-    for(auto& i: substrings_stm){
-      shift.at(0) = 0;
-      shift.at(1) = 0;
-      shift.at(2) = 0;
-      showNodesByName(node,i,kFALSE, 0, trans, beamlineholder, maxlevel, level, false, false, shift, false, false, drawconfigf.getInt("BLColor"));
+    /*if(geomOpt.showST and !geomOpt.extracted){
```

Comment:
What is the plan for this commented code?

Reply:
As per the note, I'm not sure how to find the IPA/OPA from the GDML, its nested somewhere weirdly...I want to figure it out, but it comes next on my list after the CRV